packages/beacon-node/src/sync/sync.ts

```diff
@@ -196,36 +199,48 @@ export class BeaconSync implements IBeaconSync {
     const state = this.state; // Don't run the getter twice
 
     // We have become synced, subscribe to all the gossip core topics
-    if (
-      state === SyncState.Synced &&
-      !this.network.isSubscribedToGossipCoreTopics() &&
-      this.chain.clock.currentEpoch >= MIN_EPOCH_TO_START_GOSSIP
-    ) {
-      this.network
-        .subscribeGossipCoreTopics()
-        .then(() => {
-          this.metrics?.syncSwitchGossipSubscriptions.inc({action: "subscribed"});
-          this.logger.info("Subscribed gossip core topics");
-        })
-        .catch((e) => {
-          this.logger.error("Error subscribing to gossip core topics", {}, e);
-        });
+    if (state === SyncState.Synced && this.chain.clock.currentEpoch >= MIN_EPOCH_TO_START_GOSSIP) {
+      if (!this.network.isSubscribedToGossipCoreTopics()) {
+        this.network
+          .subscribeGossipCoreTopics()
+          .then(() => {
+            this.metrics?.syncSwitchGossipSubscriptions.inc({action: "subscribed"});
+            this.logger.info("Subscribed gossip core topics");
+          })
+          .catch((e) => {
+            this.logger.error("Error subscribing to gossip core topics", {}, e);
+          });
+      }
```

Comment:
Can we rebundle this if statement as before? If you need you can re-order the conditions
```suggestion
    if (
      state === SyncState.Synced &&
      this.chain.clock.currentEpoch >= MIN_EPOCH_TO_START_GOSSIP &&
      !this.network.isSubscribedToGossipCoreTopics()
    ) {
```

Reply:
yes we can do it but then the below `this.unknownBlockSync.subscribeToNetwork();` has to depend on `this.network.isSubscribedToGossipCoreTopics()`. Logically it's the same but it'll make it hard to reason about later.